src/lib.rs

```diff
@@ -1,18 +1,146 @@
+    };
+
+    Some(cet)
+}
+
+//TODO is fund_tx_id sufficient, or should fund_vout be an argument as well
+pub fn create_cet(local_fund_pubkey: PublicKey,
+                  local_sweep_pubkey: PublicKey,
+                  remote_sweep_pubkey: PublicKey,
+                  remote_final_address: Address,
+                  oracle_pubkey: PublicKey,
+                  oracle_r_points: &[PublicKey],
+                  messages: &[String],
+                  delay: i64,
+                  local_payout: u64,
+                  remote_payout: u64,
+                  maturity_time: u32,
+                  fund_tx_id: TxIn
+) -> Option<Transaction> {
+
+    let cet_script = create_cet_redeem_script(
+        local_fund_pubkey,
+        local_sweep_pubkey,
+        remote_sweep_pubkey,
+        oracle_pubkey,
+        oracle_r_points,
+        messages, 
+        delay
+    );
+
+    create_cet_transaction(
+        cet_script.as_bytes(),
+        remote_final_address,
+        local_payout,
+        remote_payout,
+        fund_tx_id,
+        maturity_time)
+}
+
+pub fn create_cet_redeem_script(local_fund_pubkey: PublicKey,
+                                local_sweep_pubkey: PublicKey,
+                                remote_sweep_pubkey: PublicKey,
+                                oracle_pubkey: PublicKey,
+                                oracle_r_points: &[PublicKey],
+                                messages: &[String],
+                                delay: i64) -> Script {
+
+    let s = Secp256k1::signing_only();
+    let local_sweep_hash = sha256d::Hash::hash(local_sweep_pubkey.to_string().as_bytes());
+    let sk_local_sweep = SecretKey::from_slice(&local_sweep_hash).unwrap();
+    let pk_local_sweep = PublicKey::from_secret_key(&s, &sk_local_sweep);
+
+    let combine_pubkey = get_committed_key(oracle_pubkey, oracle_r_points, messages)
+        .combine(&local_fund_pubkey).unwrap()
+        .combine(&pk_local_sweep).unwrap();
+
+    Builder::new()
+        .push_opcode(opcodes::all::OP_IF)
+        .push_slice(combine_pubkey.to_string().as_bytes())
+        .push_opcode(opcodes::all::OP_ELSE)
+        .push_int(delay)
+        .push_opcode(opcodes::all::OP_CHECKSIGVERIFY)
+        .push_opcode(opcodes::all::OP_DROP)
+        .push_slice(remote_sweep_pubkey.to_string().as_bytes())
+        .push_opcode(opcodes::all::OP_ENDIF)
+        .push_opcode(opcodes::all::OP_CHECKSIG)
+        .into_script()
+}
 
 pub fn create_funding_transaction(local_fund_pubkey: PublicKey,
-                              remote_fund_pubkey: PublicKey,
-                              output_amount: u64,
-                              local_inputs: Vec<TxIn>,
-                              remote_inputs: Vec<TxIn>,
-                              local_change_output: Vec<TxOut>,
-                              remote_change_output: Vec<TxOut>) -> Transaction {
+                                  remote_fund_pubkey: PublicKey,
```

Comment:
It would be nice to use some kind of auto-formatter to avoid this kind of cosmetic changes across commits.

Reply:
Indeed.  Vim gives me issues some times with formatting.  I should put more work into a vim setting that doesn't change.